sherpa/data.py

```diff
@@ -487,14 +492,45 @@ def apply(self, array):
             raise DataErr('mismatch', 'mask', 'data array')
         return array[self.mask]
 
-    def notice(self, mins, maxes, axislist, ignore=False):
+    def notice(self, mins, maxes, axislist, ignore=False, integrated=False):
+        """Select a range.
+
+        Parameters
+        ----------
+        mins : sequence of values
+           The minimum value of the valid range (elements may be
+           None).  When not None, it is treated as an inclusive limit,
+           so points >= min are included.
+        maxes : sequence of values
+           The maximum value of the valid range (elements may be
+           None).  When not None, it is treated as an inclusive limit,
+           so points <= max are included, when upper is True,
+           otherwise points < max are included.
+        axislist: sequence of arrays
+           The axis to apply the range to. There must be the same
+           number of elements in mins, maxes, and axislist.  The
+           number of elements of each element of axislist must also
+           agree (the cell values do not need to match).
```

Comment:
I know @DougBurke is documenting existing code, and not responsible for the design of these parameters. Unfortunately, I don't know what they are supposed to be and this explanation does not make me much wiser. I have the same number of elements in min and axislist, and the number of elements of each element also has to agree but the values do not. I think an example would help greatly to clearify. Something like

```
notice([1,4], [2,7], [[2,3], [0,2]])
```
would then apply the limits 1..2 to axis 2 and 3 and the limits 4..7 to axis 0 and 2? But is that interpretation is correct and a sequence of values is simple like calling notive several times on each element in that sequence, then why does the number of elements in axis list be the same for each element? Why not apply one range to axis 0 and another one to axis 2 and 3? That makes no sense, so apparently I don't understand either the format or the intention or both.

Reply:
As I said, I don't think this is really user actionable, it's more for any poor soul that wants to write their own data class.